crates/re_types_builder/src/codegen/cpp/mod.rs

```diff
@@ -686,6 +849,133 @@ fn fill_arrow_array_builder_method(
     }
 }
 
+fn component_to_data_cell_method(
+    type_ident: &Ident,
+    hpp_includes: &mut Includes,
+    cpp_includes: &mut Includes,
+) -> Method {
+    hpp_includes.local.insert("../data_cell.hpp".to_owned());
+    cpp_includes.local.insert("../rerun.hpp".to_owned()); // ipc_from_table
+    cpp_includes.system.insert("arrow/api.h".to_owned());
+
+    let todo_pool = quote_comment("TODO(andreas): Allow configuring the memory pool.");
+
+    Method {
+        docs: format!("Creates a Rerun DataCell from an array of {type_ident} components.").into(),
+        declaration: MethodDeclaration {
+            is_static: true,
+            return_type: quote! { arrow::Result<rr::DataCell> },
+            name_and_parameters: quote! {
+                to_data_cell(const #type_ident* components, size_t num_components)
```

Comment:
The naming is quite unfortunate here :D 

If I understand correctly, you mean "component" as in "an element in an array", but that's very confusing given the context :grimacing: 

```suggestion
to_data_cell(const #type_ident* instances, size_t num_instances)
```

Reply:
oh true, I'm too deep in already 😄